src/ring_theory/finiteness.lean

```diff
@@ -90,7 +90,20 @@ variables (R)
 instance self : finite R R :=
 ⟨⟨{1}, by simpa only [finset.coe_singleton] using ideal.span_singleton_one⟩⟩
 
-variables {R}
+variable (M)
+
+lemma of_restrict_scalars_finite [module A M] [is_scalar_tower R A M] [hM : finite R M] :
```

Comment:
If I make this an instance I get a "maximum class-instance resolution depth has been reached" error in at line 422. So I kept it a lemma.

Reply:
You should make this an instance on `restrict_scalars R A M`

Reply:
Can you clarify a little bit? I don't know what you mean by "an instance on `something`".

Reply:
@riccardobrasca From `finite A M`, lean cannot figure out which `R` to look for. So it goes crazy. If on the other hand `R` can be inferred from some special case of `M`, then it's fine to make it an instance.

In particular, for this lemma it is important that `R` is an explicit variable. Because Lean will almost never be able to figure it out itself, so the user must almost always provide it anyway. So it better be explicit.